src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetriesBehavior.cs

```diff
@@ -42,7 +42,8 @@ public override void Invoke(Context context, Action next)
                 {
                     var receiveAddress = PipelineInfo.PublicAddress;
 
-                    var messageToRetry = new OutgoingMessage(context.GetPhysicalMessage().Id, message.Headers, message.Body);
+                    message.RevertToOriginalBodyIfNeeded();
+                    var messageToRetry = new OutgoingMessage(message.Id, message.Headers, message.Body);
```

Comment:
It should be the same? (he just pulled the variable up so `message = context.GetPhysicalMessage()` )